src/hooks/auditoriumSections.ts

```diff
@@ -0,0 +1,134 @@
+import { User } from "types/User";
+
+import { setGridData } from "api/profile";
+
+import {
+  currentAuditoriumSectionsSelector,
+  currentAuditoriumSectionsByIdSelector,
+} from "utils/selectors";
+import { WithId } from "utils/id";
+import { getPositionHash } from "utils/auditorium";
+
+import { useSelector } from "./useSelector";
+import { useFirestoreConnect, isLoaded } from "./useFirestoreConnect";
+import { useRecentVenueUsers } from "./users";
+import { useUser } from "./useUser";
+import { useCallback, useMemo } from "react";
+
+const useConnectAuditoriumSections = (venueId?: string) => {
+  useFirestoreConnect(() => {
+    if (!venueId) return [];
+
+    return [
+      {
+        collection: "venues",
+        doc: venueId,
+        subcollections: [{ collection: "sections" }],
+        storeAs: "currentAuditoriumSections",
+      },
+    ];
```

Comment:
Does this need to be wrapped in an array? Can we just return the object directly?

Reply:
I slightly recall the convo about this topic and I AFAIR the internal useFirebaseConnect doesn't like this kind of typing

Reply:
I think the case you're remembering is either wrong, or was fixed since what you remember:

https://github.com/sparkletown/sparkle/blob/staging/src/hooks/useFirestoreConnect.ts#L90-L97

`useFirestoreConnect` can accept both an `undefined` and a `[]` param. So my suggestion will work.

Reply:
Oh... it's in a function.. yeah.. you're correct then. My bad.

I think I look at `useCallback` / `useEffect` / etc so often that my mind tunes out the `() =>` prefix as irrelevant noise.